ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouter.java

```diff
@@ -49,7 +49,7 @@
  * Streaming, non-blocking router implementation for Ambry.
  */
 class NonBlockingRouter implements Router {
-  static final AtomicInteger currentOperationsCount = new AtomicInteger(0);
+  private final AtomicInteger currentOperationsCount = new AtomicInteger(0);
```

Comment:
Is it better to use a metric counter instead of AtomicInteger?  Do we emit metrics for the number of operations?

Reply:
I agree that the number of operation count in the router should be a non-static.  It seems like this router is a singleton, it's a good candidate for using dependency injection.

Reply:
yes its a songleton and injection would make the code so much cleaner